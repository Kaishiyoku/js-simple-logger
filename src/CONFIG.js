const CONFIG = {
    defaultDateFormat: 'HH:mm:ss.SSS',
    localStorageKeys: {
        dateFormat: 'kaishiyoku.simple-logger.dateFormat',
    },
};

export default CONFIG;
