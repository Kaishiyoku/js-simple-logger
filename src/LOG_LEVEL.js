/* eslint-disable sort-keys */
import deepFreeze from 'deep-freeze-node';

const LOG_LEVEL = deepFreeze({
    TRACE: 0,
    DEBUG: 1,
    INFO: 2,
    LOG: 3,
    WARN: 4,
    ERROR: 5,
});

export default LOG_LEVEL;
