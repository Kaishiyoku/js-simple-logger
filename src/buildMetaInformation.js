export default function buildMetaInformation(minimumLogLevel, logLevel, dateFormat, context) {
    return {
        context,
        dateFormat,
        logLevel,
        minimumLogLevel,
    };
}
