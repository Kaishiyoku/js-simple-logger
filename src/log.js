import withContext from './withContext';
import withLogLevel from './withLogLevel';
import compose from 'ramda/es/compose';
import withTimestamp from './withTimestamp';
import withMinimumLogLevel from './withMinimumLogLevel';

const log = (handler) => (metaInformation) => (...messages) => {
    const withLogLevelAndContext = compose(withTimestamp(metaInformation), withLogLevel(metaInformation), withContext(metaInformation));

    return withMinimumLogLevel(metaInformation)(() => handler(...withLogLevelAndContext(messages)));
};

export default log;
