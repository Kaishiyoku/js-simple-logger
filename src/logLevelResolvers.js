/* eslint-disable no-console */
import log from 'log';

const logLevelResolvers = {
    debug: log(console.debug),
    error: log(console.error),
    info: log(console.info),
    log: log(console.log),
    trace: log(console.trace),
    warn: log(console.warn),
};

export default logLevelResolvers;
