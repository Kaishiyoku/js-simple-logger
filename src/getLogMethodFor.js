import logLevelResolvers from './logLevelResolvers';

export default function getLogMethodFor(context, logLevel) {
    return logLevelResolvers[logLevel](context);
}
