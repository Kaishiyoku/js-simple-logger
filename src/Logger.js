/* eslint-disable sort-keys */
import getLogMethodFor from './helpers/getLogMethodFor';
import CONFIG from './data/CONFIG';
import LOG_LEVEL from './data/LOG_LEVEL';
import defaultTo from 'ramda/es/defaultTo';
import compose from 'ramda/es/compose';
import partialRight from 'ramda/es/partialRight';
import logFormatter from './helpers/logFormatter';
import Cacher from './helpers/Cacher';

const formatterFnCache = new Cacher(logFormatter);

export function setDateFormat(format) {
    localStorage.setItem(CONFIG.localStorageKeys.dateFormat, format);
}

export function getDateFormat() {
    const defaultToDateFormat = defaultTo(CONFIG.defaultDateFormat);

    return defaultToDateFormat(localStorage.getItem(CONFIG.localStorageKeys.dateFormat));
}

export function setMinimumLogLevel(logLevel) {
    localStorage.setItem(CONFIG.localStorageKeys.minimumLogLevel, logLevel);
}

export function getMinimumLogLevel() {
    const defaultToMinimumLogLevel = defaultTo(CONFIG.defaultMinimumLogLevel);

    return defaultToMinimumLogLevel(localStorage.getItem(CONFIG.localStorageKeys.minimumLogLevel));
}

export const getLogLevels = () => LOG_LEVEL;

export const getMetaInformationFor = (logLevel, context) => {
    return {
        context,
        dateFormat: getDateFormat(),
        logLevel,
        minimumLogLevel: getMinimumLogLevel(),
    };
};

export const setFormatter = (fn) => {
    formatterFnCache.set(fn);
};

export const getLogger = (context = null) => {
    const getLogMethodWithMetaInformationFor = partialRight(compose(getLogMethodFor(formatterFnCache.get()), getMetaInformationFor), [context]);

    return {
        trace: getLogMethodWithMetaInformationFor(LOG_LEVEL.TRACE),

        debug: getLogMethodWithMetaInformationFor(LOG_LEVEL.DEBUG),

        table: getLogMethodWithMetaInformationFor(LOG_LEVEL.TABLE),

        info: getLogMethodWithMetaInformationFor(LOG_LEVEL.INFO),

        log: getLogMethodWithMetaInformationFor(LOG_LEVEL.LOG),

        warn: getLogMethodWithMetaInformationFor(LOG_LEVEL.WARN),

        error: getLogMethodWithMetaInformationFor(LOG_LEVEL.ERROR),

        getContext() {
            return context;
        },
    };
};
