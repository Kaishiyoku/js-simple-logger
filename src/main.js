import {getLogger, setDateFormat, setMinimumLogLevel, setFormatter, getLogLevels} from './Logger';
import formatLogLevel from './helpers/formatLogLevel';

export {
    getLogger,
    setDateFormat,
    setMinimumLogLevel,
    setFormatter,
    getLogLevels,
    formatLogLevel,
};
