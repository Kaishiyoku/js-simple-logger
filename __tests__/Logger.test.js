/* eslint-disable no-console */
import MockDate from 'mockdate';
import {getLogLevels, getLogger, setMinimumLogLevel} from '../src/Logger';

describe('Logger', () => {
    MockDate.set('3/10/2017 15:05:20.389');

    const sampleData = [
        {name: 'A', value: 1},
        {name: 'B', value: 2},
        {name: 'C', value: 3},
        {name: 'D', value: 4},
        {name: 'E', value: 5},
    ];

    it('should log a simple message properly', () => {
        setMinimumLogLevel(getLogLevels().TRACE);
        const logger = getLogger();

        expect(logger.trace('Trace message')).toMatchSnapshot();
        expect(logger.debug('Debug message')).toMatchSnapshot();
        expect(logger.table(sampleData)).toEqual(sampleData);
        expect(logger.info('Info message')).toMatchSnapshot();
        expect(logger.log('Log message')).toMatchSnapshot();
        expect(logger.warn('Warn message')).toMatchSnapshot();
        expect(logger.error('Error message')).toMatchSnapshot();
    });

    it('should log with a context', () => {
        const logger = getLogger('main');

        expect(logger.trace('Trace message')).toMatchSnapshot();
        expect(logger.debug('Debug message')).toMatchSnapshot();
        expect(logger.table(sampleData)).toEqual(sampleData);
        expect(logger.info('Info message')).toMatchSnapshot();
        expect(logger.log('Log message')).toMatchSnapshot();
        expect(logger.warn('Warn message')).toMatchSnapshot();
        expect(logger.error('Error message')).toMatchSnapshot();
    });
});
