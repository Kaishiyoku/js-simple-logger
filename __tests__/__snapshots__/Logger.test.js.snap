// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger should log a simple message properly 1`] = `"15:05:20.389 [TRACE] Trace message"`;

exports[`Logger should log a simple message properly 2`] = `"15:05:20.389 [DEBUG] Debug message"`;

exports[`Logger should log a simple message properly 3`] = `"15:05:20.389 [INFO]  Info message"`;

exports[`Logger should log a simple message properly 4`] = `"15:05:20.389 [LOG]   Log message"`;

exports[`Logger should log a simple message properly 5`] = `"15:05:20.389 [WARN]  Warn message"`;

exports[`Logger should log a simple message properly 6`] = `"15:05:20.389 [ERROR] Error message"`;

exports[`Logger should log with a context 1`] = `"15:05:20.389 [TRACE] [main]               Trace message"`;

exports[`Logger should log with a context 2`] = `"15:05:20.389 [DEBUG] [main]               Debug message"`;

exports[`Logger should log with a context 3`] = `"15:05:20.389 [INFO]  [main]               Info message"`;

exports[`Logger should log with a context 4`] = `"15:05:20.389 [LOG]   [main]               Log message"`;

exports[`Logger should log with a context 5`] = `"15:05:20.389 [WARN]  [main]               Warn message"`;

exports[`Logger should log with a context 6`] = `"15:05:20.389 [ERROR] [main]               Error message"`;
